tests/test_dle.py

```diff
@@ -523,6 +523,33 @@ def test_scheduling(self):
         assert not iterations[3].is_Affine
         assert 'schedule(dynamic,chunk_size)' in iterations[3].pragmas[0].value
 
+    @pytest.mark.parametrize('so', [0, 1, 2])
+    @pytest.mark.parametrize('dim', [1, 2])
+    def test_array_reduction(self, so, dim):
+        grid = Grid(shape=(3, 3, 3))
+        d = grid.dimensions[dim]
+
+        f = Function(name='f', shape=(3,), dimensions=(d,), grid=grid, space_order=so)
+        u = TimeFunction(name='u', grid=grid)
+        op = Operator(Inc(f, u + 1), opt=('openmp', {'par-collapse-ncores': 1}))
+
+        iterations = FindNodes(Iteration).visit(op)
+        if so > 0:
+            assert ("reduction(+:f[%s + %s:%s + %s])" % (d.symbolic_min, so,
+                                                         d.symbolic_max, so+1)
+                    in iterations[1].pragmas[0].value)
+        else:
+            assert ("reduction(+:f[%s:%s + 1])" % (d.symbolic_min, d.symbolic_max)
+                    in iterations[1].pragmas[0].value)
+
+        try:
+            op(time_M=1)
+            assert np.allclose(f.data, 18)
+        except:
```

Comment:
this is flawed, because the assert failing would make you fall into the except branch

Reply:
Hum indeed, just `op(time_M=1)` should be in it and assert below it